File: checker/randomPlayers/RandomPlayer.java
import java.util.Scanner;

public class RandomPlayer {

   private static final int ME = 0;
   private static final int OPPONENT = 1;

	public static void main(String[] args){
		int turn = ME;
		
		if (args.length != 1){
			return;
      }
		if (args[0].equals("second")){
			turn = OPPONENT;
		}
		
      int message;
		Scanner s = new Scanner(System.in);

      /* first of all, get the difficulty settings from the sever */
      message = s.nextInt();

      /* then play randomly */
		for (;;){
			if (turn == ME){
				System.out.println((((int)(Math.random() * 100)) % 9) + "\n");
				System.out.flush();
			} else {
				message = s.nextInt();
            if (message == 10 || message == 11 || message == 12) {
               return;
            } else {
               /* ... */
            }
			}
			turn = (turn == ME) ? OPPONENT : ME;
		}
	}
}

File: checker/randomPlayers/randomPlayer.cpp
#include <iostream>
#include <cstdlib>
#include <cmath>
#include <ctime>
#include <cstring>

#define ME 0
#define OPPONENT 1
#define NCOLS 9
#define UNDEF -1

int main(int argc, char * argv[])
{
   /* check that the program received precisely one argument */
   if (argc != 2) {
      return EXIT_FAILURE;
   }

   /* default to the first player, but check if you should be second */
   int turn = ME;
   if (strcmp(argv[1],"second") == 0){
      turn = OPPONENT;
   }

   int message;
   /* first of all, get the information about the difficulty setting */
   std::cin >> message;

   /* then play randomly */
   srand(time(NULL));
   for (;;){
      if (turn == ME){
         std::cout << (rand() % NCOLS) << std::endl;
         std::cout.flush();
      } else {
         std::cin >> message;
         if (message == 10 || message == 11 || message == 12){
            return EXIT_SUCCESS;
         } else {
            /* ... */
         }
      }
      turn = (turn == ME) ? OPPONENT : ME;
   }

   /* finish up */
   return EXIT_SUCCESS;
}

